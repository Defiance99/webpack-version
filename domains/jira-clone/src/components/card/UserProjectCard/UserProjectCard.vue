<script setup lang="ts">
import { PropType, defineProps } from 'vue';
import type UserProject from '@/interfaces/UserProject.interface';

defineProps({
  project: {
    type: Object as PropType<UserProject>,
    required: true,
  },
});
</script>

<template>
  <div class="user-project-card">
    <div>
      <a class="project-name-block mb-3">
        <span class="project-icon-wrapper">
          <JIcon
            icon="projectWallet"
            container-size
          />
        </span>
        <p class="project-name text-ellipsis text-subtitle-4">
          {{ project.name }}
        </p>
        <p class="text-body-5 text-ellipsis">
          {{ project.shortInfo }}
        </p>
      </a>
      <p class="title-links-block text-body-5 mb-1">
        Short links
      </p>
      <a class="link-block j-hover-effect j-active-effect">
        <p class="link-name text-body-5">
          Total tasks
        </p>
        <JChip>
          <span class="text-body-6">{{ project.totalTasks }}</span>
        </JChip>
      </a>
      <a
        class="link-block j-hover-effect j-active-effect"
      >
        <p class="link-name text-body-5">
          My tasks
        </p>
        <JChip>
          <span class="text-body-6">{{ project.totalTasks }}</span>
        </JChip>
      </a>
      <a class="link-block j-hover-effect j-active-effect">
        <p class="link-name text-body-5">
          Complete task
        </p>
        <JChip>
          <span class="text-body-6">{{ project.totalTasks }}</span>
        </JChip>
      </a>
    </div>
  </div>
</template>

<style lang="scss" scoped>
.user-project-card {
  position: relative;
  padding: 10px 10px 10px 16px;
  width: 244px;
  box-shadow: 0 1px 2px rgba(9, 30, 66, 0.4);
  border-left: 20px solid #FFE380;
  border-radius: 3px;
  background: #fff;
  transition: 120ms;

  &:hover {
    box-shadow: 0 1px 5px rgba(9, 30, 66, 0.4);
  }
}

.link-block {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 2px;
  padding: 4px;
  cursor: pointer;
  border-radius: 3px;
}

.project-name-block {
  display: flex;
  flex-direction: column;
  position: relative;
  padding: 0 4px;
  cursor: pointer;

  &:hover {
    .project-name {
      text-decoration: underline;
    }
  }
}

.title-links-block {
  color: #6B778C;
  text-transform: uppercase;
  padding: 0 4px;
}

.project-icon-wrapper {
  display: flex;
  position: absolute;
  top: 7px;
  left: -28px;
  width: 24px;
  height: 24px;
  overflow: hidden;
  border-radius: 3px;
}
</style>
