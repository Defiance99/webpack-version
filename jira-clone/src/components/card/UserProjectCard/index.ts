import UserProjectCard from './UserProjectCard.vue';

export { UserProjectCard };
export default UserProjectCard;
