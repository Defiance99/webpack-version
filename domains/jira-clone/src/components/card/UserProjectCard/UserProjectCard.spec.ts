import { mount } from '@vue/test-utils';
import { UserProject } from '@/interfaces/UserProject.interface';
import UserProjectCard from './index';

describe('UserProjectCard.vue', () => {
  test('renders correctly', () => {
    const userProject: UserProject = {
      id: 1,
      name: 'Test project',
      totalTasks: 16,
      shortInfo: 'Software product',
    };

    const projectCardWrapper = mount(UserProjectCard, {
      props: {
        project: userProject,
      },
    } as any);

    expect(projectCardWrapper.element).toMatchSnapshot();
  });
});
