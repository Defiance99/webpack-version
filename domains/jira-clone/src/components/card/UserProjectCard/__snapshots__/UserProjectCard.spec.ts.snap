// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProjectCard.vue renders correctly 1`] = `
<div
  class="user-project-card"
>
  <div>
    <a
      class="project-name-block mb-3"
    >
      <span
        class="project-icon-wrapper"
      >
        <!---->
      </span>
      <p
        class="project-name text-ellipsis text-body-2"
      >
        Test project
      </p>
      <p
        class="text-body-5 text-ellipsis"
      >
        Software product
      </p>
    </a>
    <p
      class="title-links-block text-body-5 mb-1"
    >
       Short links 
    </p>
    <a
      class="link-block hover-effect"
    >
      <p
        class="link-name text-body-5"
      >
         Total tasks 
      </p>
      <div
        class="j-chip size-s rounded-xs"
      >
        
        <span
          class="text-body-6"
        >
          16
        </span>
        
      </div>
    </a>
    <a
      class="link-block hover-effect"
    >
      <p
        class="link-name text-body-5"
      >
         My tasks 
      </p>
      <div
        class="j-chip size-s rounded-xs"
      >
        
        <span
          class="text-body-6"
        >
          16
        </span>
        
      </div>
    </a>
    <a
      class="link-block hover-effect"
    >
      <p
        class="link-name text-body-5"
      >
         Complete task 
      </p>
      <div
        class="j-chip size-s rounded-xs"
      >
        
        <span
          class="text-body-6"
        >
          16
        </span>
        
      </div>
    </a>
  </div>
</div>
`;
